src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -76,7 +77,7 @@ class IndexLogEntryTest extends HyperspaceSuite with SQLHelper {
          |{
          |  "name" : "indexName",
          |  "derivedDataset" : {
-         |    "type" : "com.microsoft.hyperspace.index.CoveringIndex",
+         |    "type" : "com.microsoft.hyperspace.index.types.covering.CoveringIndex",
```

Comment:
Can we do `types.CoveringIndex` instead of `types.covering.coveringIndex`? 
hmm.. is that for another type of covering index?

Reply:
I'd like to give each index type their own package, like how Spark put data source implementations in the `org.apache.spark.sql.execution.datasources.{parquet, orc, json, csv}` packages.

If this seems verbose, then how about `com.microsoft.hyperspace.indexes.covering`? That is, `com.microsoft.hyperspace.index` is for index-related stuff, and `com.microsoft.hyperspace.indexes` is for index implementations.

Note that data skipping indexes will be put in the `dataskipping` package.

Reply:
I pointed it out since the package string will be shown in index log entry. I think Option 3 looks good because it's shorter.
cc @imback82 @paryoja